### DISCUSSION TITLE
Add Client Role Attributes to JWT

### DISCUSSION THREAD

**Discussion by ChristianRaoulis** [keycloak/keycloak#30297] [Q&A]
Hey Guys,

I got an OIDC Client with some Client Roles. Those client roles got attributes which are important for the application that uses the client for authentication. 

I'm trying to concat all attributes (from groups, roles, client roles and users) into a field in my jwt. I tried using a user attribute mapper but the client role attribute won't get added to the jwt.

Is there a way to add those client role attributes to the jwt using a mapper?

**Comment by robshep** [comment]
Did you find an solution for this?

**Reply by Charismara** [reply]
Sadly not

**Reply by robshep** [reply]
Sorry to hear that.
